### PR TITLE
Add get-global-config s3 command

### PR DESCRIPTION
**Pull request by jasonblanchard**

### CODE REVIEW

File: cmd/getGlobalConfigS3_test.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+	. "github.com/smartystreets/goconvey/convey"
+	"github.com/spf13/viper"
+)
+
+func TestGetGlobalConfigS3(t *testing.T) {
+	Convey("wrap", t, func() {
+		viper.Reset()
+
+		Convey("shows usage if passed --help", func() {
```
Comment (jasonblanchard) [inline cmd/getGlobalConfigS3_test.go:16]: This is a pretty lame-o test, basically just makes sure the command exists.

File: truss/getGlobalConfigS3.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+type GetGlobalConfigS3Input struct {
+	Bucket string
+	Region string
+	Key    string
+	Role   string
+	Dir    string
+}
+
+// GetGlobalConfigS3 fetch global config from S3 and put it in home dir
+func GetGlobalConfigS3(input *GetGlobalConfigS3Input) error {
```
Comment (jasonblanchard) [inline truss/getGlobalConfigS3.go:25]: Not quite sure how to test this without mocking the AWS stuff out so much that it's not even really testing anything.

I suppose I could do something like this https://github.com/instructure-bridge/truss-cli/blob/master/truss/getKubeconfigS3_test.go#L12. @jgramoll how are you handling those `TEST_*` credentials?

File: cmd/getGlobalConfigS3.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+	},
+}
+
+func init() {
+	getGlobalConfigCmd.AddCommand(getGlobalConfigS3Cmd)
+	getGlobalConfigS3Cmd.Flags().StringP("bucket", "b", "truss-cli-global-config", "S3 bucket that contains your .truss.yaml file")
+	getGlobalConfigS3Cmd.Flags().StringP("key", "k", ".truss.yaml", "Name of the .truss.yaml file in the bucket")
+	getGlobalConfigS3Cmd.Flags().StringP("region", "r", "us-east-2", "Region of S3 bucket that contains your .truss.yaml file")
+	getGlobalConfigS3Cmd.Flags().StringP("role", "u", "", "Role with access to the S3 bucket with your .truss.yaml file")
+}
```
Comment (jgramoll) [inline cmd/getGlobalConfigS3.go:52]: Do we want to add an --out or --file to override where to save it

Comment (jasonblanchard) [reply cmd/getGlobalConfigS3.go:52]: Oh yeah, good call. I can add that

File: cmd/getGlobalConfigS3.go [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+Uses S3 under the hood so it only works if you have AWS credentials set in your shell. Use the --role flag if you need to specify an AWS role ARN:
+
+	$ truss get-global-config s3 --role arn:aws:iam::127178877223:role/xacct/ops-admin
+		`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		bucket, err := cmd.Flags().GetString("bucket")
+		key, err := cmd.Flags().GetString("key")
+		region, err := cmd.Flags().GetString("region")
+		role, err := cmd.Flags().GetString("role")
+		dir, err := cmd.Flags().GetString("out")
```
Comment (jasonblanchard) [inline cmd/getGlobalConfigS3.go:27]: @jgramoll updated with your suggestion of a configurable `--out` param

File: cmd/getGlobalConfigS3.go [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+	},
+}
+
+func init() {
+	getGlobalConfigCmd.AddCommand(getGlobalConfigS3Cmd)
+	getGlobalConfigS3Cmd.Flags().StringP("bucket", "b", "truss-cli-global-config", "S3 bucket that contains your .truss.yaml file")
+	getGlobalConfigS3Cmd.Flags().StringP("key", "k", ".truss.yaml", "Name of the .truss.yaml file in the bucket")
+	getGlobalConfigS3Cmd.Flags().StringP("region", "r", "us-east-2", "Region of S3 bucket that contains your .truss.yaml file")
+	getGlobalConfigS3Cmd.Flags().StringP("role", "u", "", "Role with access to the S3 bucket with your .truss.yaml file")
+	getGlobalConfigS3Cmd.Flags().StringP("out", "o", "", "Output directory where the .truss.yaml file is written")
```
Comment (jasonblanchard) [inline cmd/getGlobalConfigS3.go:63]: Actually, wait... let me set the default here instead of down below

File: cmd/getGlobalConfigS3.go [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+		if err != nil {
+			return err
+		}
+
+		return nil
+	},
+}
+
+func init() {
+	home, err := homedir.Dir()
```
Comment (jasonblanchard) [inline cmd/getGlobalConfigS3.go:50]: Is this (and the error handling) a bad idea in here?

File: truss/getGlobalConfigS3.go (hunk 2 above)
Comment (jgramoll) [reply truss/getGlobalConfigS3.go:24]: just using a .envrc with
```export TEST_S3_BUCKET=truss-kubeconfig-us-east-2
export TEST_AWS_ROLE=arn:aws:iam::127178877223:role/xacct/ops-admin
export TEST_VAULT_ROLE=admin
```

**Pull request merged**